stdpopsim/slim_engine.py

```diff
@@ -1052,7 +1046,14 @@ def fix_time(event):
     # Check that drawn mutations exist for extended events that need them.
     for single_site_id in referenced_single_site_ids:
         if single_site_id not in drawn_single_site_ids:
-            raise ValueError(f"No drawn mutation for single site id {single_site_id}.")
+            raise ValueError(
+                f"No drawn mutation exists for single site id {single_site_id}."
```

Comment:
Maybe I'm wrong here but shouldn't this be
```suggestion
                f"Single site id {single_site_id} is referenced but does not exist."
```

Reply:
Nope, this is checking that a single site with a `ChangeMutationFitness` or `ConditionOnAlleleFrequency` event also has a `DrawMutation`  event.

The check that the single site exists happens earlier, here:
https://github.com/popsim-consortium/stdpopsim/blob/a0da48882a7eb69a77778a620e78447ef775f3ce/stdpopsim/slim_engine.py#L960